### PR TITLE
8269342: CICrashAt=1 does not always catch first Java method

### PR DESCRIPTION
**Pull request by dean-long**
This change reintroduces the CICountNative flag, so that -XX:+CICountNative -XX:CICrashAt=1 can reliably crash the first compiled Java method without worrying about native method wrappers and method handle intrinsics.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8269342](https://bugs.openjdk.java.net/browse/JDK-8269342): CICrashAt=1 does not always catch first Java method

### Reviewers
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to ba0569e2dab579a6b5a0e75d5d1522ad50ff3d10
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4897/head:pull/4897` \
`$ git checkout pull/4897`

Update a local copy of the PR: \
`$ git checkout pull/4897` \
`$ git pull https://git.openjdk.java.net/jdk pull/4897/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4897`

View PR using the GUI difftool: \
`$ git pr show -t 4897`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4897.diff">https://git.openjdk.java.net/jdk/pull/4897.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/compiler/ciReplay/CiReplayBase.java [hunk 1]
```diff
@@ -68,17 +68,23 @@
...
     private static int dummy;
 
+    static interface Lambda {
+        int value();
+    }
+
     public static class TestMain {
         public static void main(String[] args) {
-            for (int i = 0; i < 20_000; i++) {
+            Lambda start = () -> 0;
```
Comment (vnkozlov) [inline test/hotspot/jtreg/compiler/ciReplay/CiReplayBase.java:85]: I would add comment that you use this to trigger native compilation explicitly.

**Review by vnkozlov — APPROVED** [review]
Looks good.

**Comment by dean-long** [comment]
Thanks for the review, Vladimir.

**Review by TobiHartmann — APPROVED** [review]
Looks good.

**Comment by dean-long** [comment]
Thanks, Tobias.

**Pull request closed without merging**